orangecontrib/bio/widgets3/OWdictyExpress.py

```diff
@@ -0,0 +1,235 @@
+
+        self.experimentsWidget = QTreeWidget()
+        self.experimentsWidget.setAlternatingRowColors(True)
+        self.experimentsWidget.setRootIsDecorated(False)
+        self.experimentsWidget.setUniformRowHeights(True)
+        self.experimentsWidget.setSortingEnabled(True)
+
+        self.experimentsWidget.setItemDelegateForColumn(
+            0, gui.IndicatorItemDelegate(self, role=Qt.DisplayRole))
+
+        self.experimentsWidget.selectionModel().selectionChanged.connect(
+            self.onSelectionChanged)
+
+        self.experimentsWidget.setHeaderLabels(self.headerLabels)
+        self.mainArea.layout().addWidget(self.experimentsWidget)
+
+        if self.username and self.password:
+            self.connect()
+
+    def auth_set(self):
+        if len(self.username):
+            self.passfield.setDisabled(False)
+        else:
+            self.passfield.setDisabled(True)
+
+    def auth_changed(self):
+        self.auth_set()
+        self.connect()
+
+    def refresh(self):
+        self.reset()
+        self.load_experiments()
+
+    def reset(self):
+        self.experimentsWidget.clear()  # clear QTreeWidget
+        self.items = []
+        self.lastSelected = None
+        self.searchString = ""
+        self.handle_commit_button(False)
+
+    def search_update(self):
+        for item in self.items:
+            item.setHidden(
+                not all(s in item for s in self.searchString.split()))
+
+    def connect(self):
+        self.res = None
+        self.error(1)
+        self.reset()
+        self.handle_commit_button(False)
+        self.handle_cache_button(False)
+
+        if self.username and self.password:
+            try:
+                self.res = Genapi(self.username, self.password, self.server)
+            except Exception as e:
+                self.error(1, e.args[0])
+            else:
+                self.load_experiments()
+                self.handle_cache_button(True)
+
+    def load_experiments(self):
+        pbar = gui.ProgressBar(self, iterations=100)
+        if self.res:
+            try:
+                experiments = self.res.fetch_etc_objects(pbar.advance)  # returns a list of experiments
+            except Exception as e:
+                self.error(1, e.args[0])
+            else:
+                self.load_tree_items(experiments)
+        pbar.finish()
+
+    def load_tree_items(self, list_of_exp):
+        self.items = [CustomTreeItem(self.experimentsWidget, item) for item in list_of_exp]
+        for i in range(len(self.headerLabels)):
+            self.experimentsWidget.resizeColumnToContents(i)
+
+    def onSelectionChanged(self):
+        self.handle_commit_button(True)
+
+    def handle_commit_button(self, handle):
+        self.commit_button.setEnabled(handle)
+
+    def handle_cache_button(self, handle):
+        self.refresh_button.setEnabled(handle)
+
+    def commit(self):
+        pbar = gui.ProgressBar(self, iterations=100)
```

Comment:
```py
with ...
```

Why set to 100? If this is not some arbitrary value and this 100 must match above 100 (in `load_experiments()`), then make it a named constant.